core/domain/user_services_test.py

```diff
@@ -129,3 +133,236 @@ def test_get_user_id_from_username(self):
         # Return None for usernames which don't exist.
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
+
+
+class UpdateContributionMsecTests(test_utils.GenericTestBase):
+    """Test whether contribution date changes with publication of
+    exploration/collection and update of already published exploration/collection
+    """
+
+    EXP_ID = 'test_exp'
+    COL_ID = 'test_col'
+    COLLECTION_TITLE = 'title'
+    COLLECTION_CATEGORY = 'category'
+    COLLECTION_OBJECTIVE = 'objective'
+
+    def test_contribution_msec_updates_on_published_explorations(self):
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.admin_id, end_state_name='End')
+        self.init_state_name = exploration.init_state_name
+        rights_manager.publish_exploration(self.admin_id, self.EXP_ID)
+
+        # Test all owners and editors of exploration after publication have
+        # updated first contribution times in msecs.
+        self.assertIsNotNone(user_services.get_user_settings(self.admin_id)
+            .first_contribution_msec)
+
+        # Test editor of published exploration has updated contribution time.
+        rights_manager.release_ownership_of_exploration(self.admin_id,
+            self.EXP_ID)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        exp_services.update_exploration(
+            self.editor_id, self.EXP_ID, [{
+            'cmd': 'edit_state_property',
+            'state_name': self.init_state_name,
+            'property_name': 'widget_id',
+            'new_value': 'MultipleChoiceInput'
+            }], 'commit')
+
+        self.assertIsNotNone(user_services.get_user_settings(self.editor_id)
+            .first_contribution_msec)
+
+    def test_contribution_msec_updates_on_unpublished_explorations(self):
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
```

Comment:
I might not have caught this above, but you shouldn't really be logging in as two people at the same time -- you should logout before you log in again. Could you check that none of the other tests make this error?